configure.ac

```diff
@@ -28,12 +28,47 @@ AC_PREFIX_DEFAULT(/usr/local/joy)
 AM_INIT_AUTOMAKE([subdir-objects -Wall -Werror])
 AC_USE_SYSTEM_EXTENSIONS
 AM_MAINTAINER_MODE
+AM_PROG_AR
 #
 
 # Checks for programs.
 AC_PROG_CC
-AM_PROG_CC_C_O
+AC_PROG_CXX
 AC_PROG_RANLIB
+AC_FUNC_STRNLEN
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_TYPE_MODE_T
+AC_TYPE_SSIZE_T
+AC_TYPE_INT8_T
+AC_TYPE_INT16_T
+AC_TYPE_INT64_T
```

Comment:
Do we need AC_TYPE_INT32_T, AC_TYPE_UINT8_T ... AC_TYPE_UINT64_T?

Reply:
I will add them for completeness.

Reply:
Builds on MAC too